### ISSUE TITLE
Support Elytron OpenSSL TLSv1.3 provider with Vert.x HTTP

### ISSUE THREAD

**Issue by sberyozkin** [quarkusio/quarkus#34394]
### Description

Farah @fjuma has mentioned that the Elytron team has observed a better TLSv1.3 performance when an OpenSSL TLSv1.3 provider is selected instead of the default Java TLSv1.3 provider.
Darran @darranl has also mentioned that the related OpenSSL code is FIPS compliant.  

Letting Quarkus users choose the Elytron OpenSSL TLSv1.3 provider can help with improving the performance related to the TLSv1.3 implementation. 

CC @cescoffier @stuartwdouglas @geoand  @vietj

**Comment by cescoffier** [comment]
Hum... I'm somewhat concerned. Especially in native mode, it can become very complex.
Also, my experience with netty and OpenSSL did not generally end well.

**Comment by sberyozkin** [comment]
Hi Clement, 

>my experience with netty and OpenSSL could have been better. (blocked event loop)

Would it be correct to assume that plugging in a 3rd part provider like Elytron OpenSSL into a Netty, instead of the Netty Open SSL, it technically not possible ?

CC @fjuma

**Comment by sberyozkin** [comment]
Might be worth giving Netty and OpenSSL another try at some point

**Comment by fjuma** [comment]
Right, just to note, Netty also has its own Java implementation that makes use of JNI to invoke OpenSSL functions, similar to the WildFly OpenSSL provider so might be worth looking into that as well.

**Comment by cescoffier** [comment]
The JNI route is a few drawbacks when looking at native compilation. 

Basically you need to lock the OpenSSL location, and the host building the native and the one running it must you the same location. 

If both are RHEL, no problem. But if one is Ubuntu and then you run in a ubi container, it may not work. 

Also, neither Quarkus micro nor distroless images are shipping OpenSSL. That that would be another change to plan (and we must look at the size overhead). If it requires to use ubi minimal, the image size can become a problem.

**Comment by cescoffier** [comment]
also, the JNI route is a no go with virtual threads - it will pin the carrier thread.

**Comment by cescoffier** [comment]
I'm going to close this issue. We will integrate OpenSSL, but using Netty.
